modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2552,16 +2559,23 @@ public CodegenModel fromModel(String name, Schema schema) {
 
                     if (composed.getAnyOf() != null) {
                         m.anyOf.add(modelName);
+                        anyOfProps.add(interfaceProperty);
                     } else if (composed.getOneOf() != null) {
                         m.oneOf.add(modelName);
+                        oneOfProps.add(interfaceProperty);
                     } else if (composed.getAllOf() != null) {
                         m.allOf.add(modelName);
+                        allOfProps.add(interfaceProperty);
                     } else {
                         LOGGER.error("Composed schema has incorrect anyOf, allOf, oneOf defined: {}", composed);
                     }
                 }
             }
 
+            m.vendorExtensions.put("x-oneOf-as-properties", oneOfProps);
+            m.vendorExtensions.put("x-allOf-as-properties", allOfProps);
+            m.vendorExtensions.put("x-anyOf-as-properties", anyOfProps);
```

Comment:
I'll rename these to oneOfProps, anyOfProps and allOfProps instead after merging this PR.

Reply:
Done via https://github.com/OpenAPITools/openapi-generator/pull/10231